module/signature/signer_store.go

```diff
@@ -40,16 +39,24 @@ func (s *EpochAwareSignerStore) GetThresholdSigner(view uint64) (module.Threshol
 		return signer, nil
 	}
 
-	privDKGData, err := s.keys.RetrieveMyBeaconPrivateKey(epoch)
-	if errors.Is(err, storage.ErrNotFound) {
-		signer = NewThresholdProvider(encoding.RandomBeaconTag, nil)
-	} else if err != nil {
+	privDKGData, safe, err := s.keys.RetrieveMyBeaconPrivateKey(epoch)
+	if err != nil {
+		// there are no expected errors here
 		return nil, fmt.Errorf("could not retrieve DKG private key for epoch counter: %v, at view: %v, err: %w", epoch, view, err)
+	}
+
+	if !safe {
+		// we do not have a consistent beacon key which is safe for signing
+		// CAUTION: for now, we produce explicitly invalid signatures
+		// TODO: in Crypto V2, we will fallback to using staking signatures
+		signer = NewThresholdProvider(encoding.RandomBeaconTag, nil)
 	} else {
+		// we have a beacon key which has been explicitly marked safe for use
+		// by the DKG process
```

Comment:
The way `RetrieveMyBeaconPrivateKey` is written, `safe` is `false` iff `privDKGData` is `nil`. This means we can just call `signer = NewThresholdProvider(encoding.RandomBeaconTag, privDKGData)` in all cases.

Reply:
This also means the boolean return of `RetrieveMyBeaconPrivateKey` is not very useful. The safety check is done already within `RetrieveMyBeaconPrivateKey`.

Reply:
That happens to be true here (we could remove the conditional and behaviour would be the same), but that won't be true when this is incorporated into Crypto V2.

It would be possible to remove `safe` and say that `(nil, nil)` means we don't have a safe key, but this breaks with common convention -- in particular, if a function returns `(value, error)` and the caller observes `error==nil`, 99% of the time that means the function returned a valid `value` that can be safely used.